sample-operators/mysql-schema/src/main/java/io/javaoperatorsdk/operator/sample/MySQLSchemaReconciler.java

```diff
@@ -32,9 +33,9 @@ public MySQLSchemaReconciler() {}
     Secret secret = context.getSecondaryResource(Secret.class).orElseThrow();
     SchemaDependentResource schemaDependentResource = context.managedDependentResourceContext()
         .getDependentResource(SchemaDependentResource.class);
-    return schemaDependentResource.fetchResource(schema).map(s -> {
+    return schemaDependentResource.getResource(schema).map(s -> {
```

Comment:
Why the change here?

Reply:
It's unrelated to the problem. I think we should rather get the actual state from cache, and handle the details when to call the actual service to the dependent resource implementation.